src/main/java/com/code_intelligence/jazzer/mutation/combinator/MutatorCombinators.java

```diff
@@ -174,6 +174,57 @@ public R detach(R value) {
     };
   }
 
+  public static <T, R> SerializingMutator<R> mutateThenMapToImmutable(
+      SerializingMutator<T> mutator, Function<T, R> map, Function<R, T> inverse, String name) {
+    return new PostComposedMutator<T, R>(mutator, map, inverse) {
+      @Override
+      public R detach(R value) {
+        return value;
+      }
+
+      @Override
+      public String toString() {
+        return name;
+      }
+    };
+  }
+
+  public static SerializingMutator<Integer> mutateIntegralInRange(
```

Comment:
Since this is only an internal mutator that we use to construct other ones, we could simplify it by restricting to ranges of the form `[0, upperExclusive)`, which fits zero-based indices very well.

If you do that, you also don't need `PseudoRandom#otherInClosedRange`, but just an additional overload for `PseudoRandom#otherIndex`.